src/scripts/main.js

```diff
@@ -1,3 +1,65 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field');
+const tbody = table.querySelector('tbody');
+const appendRow = document.querySelector('.append-row');
+const removeRow = document.querySelector('.remove-row');
+const appendColumn = document.querySelector('.append-column');
+const removeColumn = document.querySelector('.remove-column');
+
+function updateButtonStates() {
+  const trs = tbody.querySelectorAll('tr');
+  const rowCount = trs.length;
+  const columnCount = trs[0].children.length;
+
+  appendRow.disabled = rowCount >= 10;
+  removeRow.disabled = rowCount <= 2;
+  appendColumn.disabled = columnCount >= 10;
+  removeColumn.disabled = columnCount <= 2;
+}
+
+appendRow.addEventListener('click', () => {
+  const trs = tbody.querySelectorAll('tr');
+
+  if (trs.length < 10) {
+    const tdsCount = trs[0].children.length;
```

Comment:
Similar to the previous comment, ensure that `trs` is not empty before accessing `trs[0]`. Add a check to handle cases where `trs.length` is 0 to prevent errors.